backends/openweathermap.org.go

```diff
@@ -82,15 +93,52 @@ func (c *openWeatherConfig) fetch(url string) (*openWeatherResponse, error) {
 	if c.debug {
 		fmt.Printf("Response (%s):\n%s\n", url, string(body))
 	}
+	if res.StatusCode != 200 {
+		err = openWeatherErrorReponseHandler(body, url)
+		return nil, err
+	} else {
+		var resp openWeatherResponse
+		if err = json.Unmarshal(body, &resp); err != nil {
+			return nil, fmt.Errorf("Unable to unmarshal response (%s): %v\nThe json body is: %s", url, err, string(body))
+		}
+		if resp.Cod != "200" {
+			return nil, fmt.Errorf("Erroneous response body: %s", string(body))
+		}
+		return &resp, nil
+	}
+}
+
+func openWeatherErrorReponseHandler(body []byte, url string) error {
+	var resp openWeatherErrorReponse
+
+	if err := resp.UnmarshalJSON(body); err != nil {
+		return fmt.Errorf("Unable to unmarshal error response (%s): %v\nThe json body is: %s", url, err, string(body))
+	}
+
+	return resp
+}
 
-	var resp openWeatherResponse
-	if err = json.Unmarshal(body, &resp); err != nil {
-		return nil, fmt.Errorf("Unable to unmarshal response (%s): %v\nThe json body is: %s", url, err, string(body))
+func (r *openWeatherErrorReponse) UnmarshalJSON(data []byte) error {
+	var raw struct {
+		Cod     interface{} `json:"cod"`
+		Message string      `json:"message"`
 	}
-	if resp.Cod != "200" {
-		return nil, fmt.Errorf("Erroneous response body: %s", string(body))
+
+	if err := json.Unmarshal(data, &raw); err != nil {
+		return err
 	}
-	return &resp, nil
+
+	switch v := raw.Cod.(type) {
```

Comment:
This is weird: can the type really be a string? Codes are usually only integers, so using a float64 here makes no sense to me. According to the docs this should always be an integer: https://openweathermap.org/api/one-call-3#errorstructure

Reply:
### String and Integer Cod Value - 
so, if you see this

- This is the API response with a success, cod is of type string
![image](https://github.com/user-attachments/assets/bdcff99e-2e24-4396-869d-cdc849ced3bd)

- This is the API response when I am tampering my API key they are sending a integer
![image](https://github.com/user-attachments/assets/6ae99fb4-490e-437e-8a10-4eaa724dbbe0)

- This is the API response when I am sending request to a wrong path (removed e from the forecast)
https://api.openweathermap.org/data/2.5/forcast?q=london&appid=06c3a7c4c72bf18af27ea371db9f7530&units=metric&lang=en
![image](https://github.com/user-attachments/assets/dcc11ae9-a86e-4e07-a267-8f4539636522)


They are sending a string in few cases like this 404 and 200 but for 400 it is an integer

Reply:
### Why Float64?

The reason why I am comparing type with float64 is because I am taking Code as a type of interface and when go unmarshals an interface value for numbers it returns the number with the type of ```float64```

Source: 
https://stackoverflow.com/questions/39152481/unmarshaling-a-json-integer-to-an-empty-interface-results-in-wrong-type-assertio